### PR TITLE
manifest: Bring Oberon PSA repo

### PR DESCRIPTION
**Pull request by Vge0rge**
With support for legacy and PSA for
nRF54L15.

### CODE REVIEW

**Comment by krish2718** [comment]
Build error is resolved, but seeing linker error with this:

```
zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_setup':
work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1261: undefined reference to `oberon_hash_setup'
zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_update':
work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1320: undefined reference to `oberon_hash_update'
zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_finish':
work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1345: undefined reference to `oberon_hash_finish'
zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_abort':
work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1369: undefined reference to `oberon_hash_abort'
collect2: error: ld returned 1 exit status
```

**Comment by Vge0rge** [comment]
> Build error is resolved, but seeing linker error with this:
> 
> ```
> zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_setup':
> work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1261: undefined reference to `oberon_hash_setup'
> zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_update':
> work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1320: undefined reference to `oberon_hash_update'
> zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_finish':
> work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1345: undefined reference to `oberon_hash_finish'
> zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: modules/nrf/subsys/nrf_security/src/libmbedcrypto.a(psa_crypto_driver_wrappers.c.obj): in function `psa_driver_wrapper_hash_abort':
> work/main/nrf/subsys/nrf_security/src/psa_crypto_driver_wrappers.c:1369: undefined reference to `oberon_hash_abort'
> collect2: error: ld returned 1 exit status
> ```

This should be fixed now.

**Review by vili-nordic — APPROVED** [review]

File: subsys/nrf_security/Kconfig [hunk 1]
```diff
@@ -70,6 +70,10 @@ config NRF_SECURITY_LEGACY_AND_PSA
 	depends on PSA_CRYPTO_DRIVER_OBERON && !PSA_CRYPTO_DRIVER_CC3XX
 	depends on NRF_CC3XX_PLATFORM || SOC_SERIES_NRF54LX
 	depends on TRUSTED_STORAGE
+	# We cannot support the encrypted backend for trusted storage in nRF54L15
+	# since it requires KMU support. The KMU driver in nRF54L15 is tightly coupled
+	# with the CRACEN PSA driver which is incombatible with legacy and PSA option.
```
Comment (krish2718) [inline subsys/nrf_security/Kconfig:75]:
```suggestion
	# with the CRACEN PSA driver which is incompatible with legacy and PSA option.
```

**Review by krish2718 — APPROVED** [review]

**Review by divyagona — APPROVED** [review]
Connectivity in 2.4Ghz and 5GHz- wpa3- successful

**Pull request merged**